core/transformations.cpp

```diff
@@ -1367,6 +1368,80 @@ void rotation3DMatrixFromIcoOrientations(const char* icoFrom, const char* icoTo,
     R = Rto * Rfrom.transpose();
 }
 
+template<typename T>
+void radialAverageNonCubic(const MultidimArray< T >& m,
+		Matrix1D< int >& center_of_rot,
+		MultidimArray< T >& radial_mean,
+		MultidimArray< int >& radial_count,
+		bool rounding)
+{
+	Matrix1D< double > idx(3);
+
+	size_t sizemax = std::max({XSIZE(m), YSIZE(m), ZSIZE(m)});
+	double scalex = XSIZE(m)/sizemax;
+	double scaley = YSIZE(m)/sizemax;
+	double scalez = ZSIZE(m)/sizemax;
+
+	// If center_of_rot was written for 2D image
+	if (center_of_rot.size() < 3)
+		center_of_rot.resize(3);
+
+	// First determine the maximum distance that one should expect, to set the
+	// dimension of the radial average vector
+	MultidimArray< int > distances(8);
+
+	const double z0 = STARTINGZ(m) - ZZ(center_of_rot);
+	const double y0 = STARTINGY(m) - YY(center_of_rot);
+	const double x0 = STARTINGX(m) - XX(center_of_rot);
+
+	const double xf = FINISHINGX(m) - XX(center_of_rot);
+	const double yf = FINISHINGY(m) - YY(center_of_rot);
+	const double zf = FINISHINGZ(m) - ZZ(center_of_rot);
+
+	distances(0) = (int) floor(sqrt(x0 * x0 + y0 * y0 + z0 * z0));
+	distances(1) = (int) floor(sqrt(xf * xf + y0 * y0 + z0 * z0));
+	distances(2) = (int) floor(sqrt(xf * xf + yf * yf + z0 * z0));
+	distances(3) = (int) floor(sqrt(x0 * x0 + yf * yf + z0 * z0));
+	distances(4) = (int) floor(sqrt(x0 * x0 + yf * yf + zf * zf));
+	distances(5) = (int) floor(sqrt(xf * xf + yf * yf + zf * zf));
+	distances(6) = (int) floor(sqrt(xf * xf + y0 * y0 + zf * zf));
+	distances(7) = (int) floor(sqrt(x0 * x0 + y0 * y0 + zf * zf));
+
+	int dim = CEIL(distances.computeMax()) + 1;
```

Comment:
I would like to see an explanation here why you need `ceil`, `+` and `rounding`.
Please add also tests for this